src/properties/columns.rs

```diff
@@ -0,0 +1,109 @@
+//! CSS properties related to Multi-column Layout.
+//! https://www.w3.org/TR/css-multicol-1/
+
+use super::{Property, PropertyId};
+use crate::context::PropertyHandlerContext;
+use crate::declaration::{DeclarationBlock, DeclarationList};
+use crate::error::{ParserError, PrinterError};
+use crate::macros::{define_shorthand, shorthand_handler};
+use crate::printer::Printer;
+use crate::targets::Browsers;
+use crate::traits::{Parse, PropertyHandler, Shorthand, ToCss};
+use crate::values::length::{IntegerOrAuto, Length, LengthOrAuto, LengthValue};
+use cssparser::*;
+use std::ops::Not;
+
+define_shorthand! {
+  /// A value for the [columns](https://www.w3.org/TR/css-multicol-1/#columns) shorthand property.
+  /// columns = <'column-width'> || <'column-count'>
+  pub struct Columns {
+    /// The column-width property.
+    width: ColumnWidth(LengthOrAuto),
+    /// The column-count property.
+    count: ColumnCount(IntegerOrAuto),
+  }
+}
+
+impl<'i> Parse<'i> for Columns {
+  fn parse<'t>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i, ParserError<'i>>> {
+    let state = input.state();
+    let with_or_count = LengthOrAuto::parse(input)?;
+
+    if let LengthOrAuto::Length(Length::Value(LengthValue::Unitless(..))) = with_or_count {
```

Comment:
Rather than parsing as a unitless length, you could instead first try parsing the column count, and if that fails, try parsing the length. When parsing shorthands where order doesn't matter, it can be useful to structure the code using a `loop`, trying to parse each property until values are found. There are many examples, [here's one](https://github.com/parcel-bundler/lightningcss/blob/865270134da6f25d66ce4a4ec0e4c4b93b90b759/src/properties/flex.rs#L81).